test/index.js

```diff
@@ -575,6 +579,159 @@ it('fails setting session key/value because of failed cache set', { parallel: fa
     });
 });
 
+it('preAuth returns 500 error if cache not ready and errorOnCacheNotReady set to default', { parallel: false }, function (done) {
+
+    var options = {
+        maxCookieSize: 0,
+        cookieOptions: {
+            password: 'password',
+            isSecure: false
+        }
+    };
+
+    var cache = require('./failing-cache.js');
+    sinon.stub(cache.prototype, 'set', function (key, value, ttl, callback) {
+
+        return callback();
+    });
+
+    var hapiOptions = {
+        cache: {
+            engine: cache
+        },
+        debug: false
+    };
+    var server = new Hapi.Server(hapiOptions);
+    server.connection();
+
+    var handlerSpy = sinon.spy();
+    server.route([
+        {
+            method: 'GET', path: '/', handler: function (request, reply) {
+
+                handlerSpy();
+                request.session.set('some', 'value');
+                return reply();
+            }
+        },
+        {
+            method: 'GET', path: '/2', handler: function (request, reply) {
+
+                handlerSpy();
+                request.session.set(45.68, '2');
+                return reply('1');
+            }
+        }
+    ]);
+
+    server.register({ register: require('../'), options: options }, function (err) {
+
+        expect(err).to.not.exist();
+        server.start(function () {
+
+            server.inject({ method: 'GET', url: '/' }, function (res) {
+
+                var header = res.headers['set-cookie'];
+                var cookie = header[0].match(/(session=[^\x00-\x20\"\,\;\\\x7F]*)/);
+
+                expect(res.statusCode).to.equal(200);
+                expect(handlerSpy.calledOnce).to.equal(true);
+
+                cache.prototype.set.restore();
```

Comment:
And for this, can we just add a public method/object on the failing cache mock object that allows us to control how `set` returns?  Again, just to avoid adding another dependency and style to the project.